app/upgrades/v12/upgrades.go

```diff
@@ -42,6 +48,34 @@ func CreateUpgradeHandler(
 			bpm.StoreConsensusParams(ctx, cp)
 		}
 
+		// Update ICA allowedMessages param
+		params := keepers.ICAHostKeeper.GetParams(ctx)
+		// Specifying the whole list instead of adding and removing. Less fragile.
+		params.AllowMessages = []string{
+			sdk.MsgTypeURL(&banktypes.MsgSend{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgDelegate{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgBeginRedelegate{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgCreateValidator{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgEditValidator{}),
```

Comment:
We also need to add staking unbond / undelegate (I forget what its called)

Reply:
added undelegate.